src/components/Promptbar/components/PromptFolders.tsx

```diff
@@ -242,26 +250,40 @@ export const PromptSection = ({
   );
 };
 
-const folderItems: FolderSectionProps<Prompt>[] = [
-  {
-    name: 'Share With Me',
-    itemFilter: SharedWithMeFilter,
-    displayRootFiles: true,
-    dataQa: 'share-with-me',
-  },
-  {
-    name: 'Pinned prompts',
-    itemFilter: PinnedItemsFilter,
-    showEmptyFolders: true,
-    openByDefault: true,
-    dataQa: 'pinned-prompts',
-  },
-];
-
 export function PromptFolders() {
+  const { t } = useTranslation(Translation.PromptBar);
+  const isFilterEmpty = useAppSelector(
+    PromptsSelectors.selectIsEmptySearchFilter,
+  );
+  const commonItemFilter = useAppSelector(PromptsSelectors.selectItemFilter);
+  const isSharingEnabled = useAppSelector((state) =>
+    SettingsSelectors.isFeatureEnabled(state, Feature.PromptsSharing),
+  );
+
+  const folderItems: FolderSectionProps<Prompt>[] = useMemo(
+    () =>
+      [
+        {
+          hidden: !isSharingEnabled || !isFilterEmpty,
+          name: t('Shared with me'),
+          itemFilter: SharedWithMeFilter,
+          displayRootFiles: true,
+          dataQa: 'share-with-me',
+        },
+        {
+          name: t('Pinned prompts'),
+          itemFilter: commonItemFilter,
+          showEmptyFolders: isFilterEmpty,
+          openByDefault: true,
+          dataQa: 'pinned-prompts',
+        },
+      ].filter(({ hidden }) => !hidden),
+    [commonItemFilter, isFilterEmpty, isSharingEnabled, t],
+  );
+
   return (
     <div
-      className="flex w-full flex-col gap-0.5 divide-y divide-gray-200 dark:divide-gray-800"
+      className="flex w-full flex-col gap-0.5 divide-y divide-gray-200 empty:hidden dark:divide-gray-800"
```

Comment:
@mikitabut ,
you can investigate code and found that it's almost impossible to check if all children are null because it's responsibility of each section to display itself